lib/3scale/client.rb

```diff
@@ -52,24 +52,37 @@ class Client
     private_constant :EXTENSIONS_HEADER
 
     def initialize(options)
-      if options[:provider_key].nil? || options[:provider_key] =~ /^\s*$/
-        raise ArgumentError, 'missing :provider_key'
+      @provider_key = options[:provider_key]
+      @service_token = options[:service_token]
+      @use_service_tokens = options[:use_service_tokens]
```

Comment:
I think that accepting `use_service_tokens` as an option is OK, but shouldn't `service_id` be required when a `service_token` is provided?